wgpu-core/src/instance.rs

```diff
@@ -457,20 +486,25 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
             _ => Vec::new(),
         };
         #[cfg(windows)]
-        let mut adapters_dx11 = if id_dx11.is_some() {
-            let mut adapters = instance.dx11.enumerate_adapters();
-            if let Some(surface) = compatible_surface {
-                adapters.retain(|a| {
-                    a.queue_families
-                        .iter()
-                        .find(|qf| qf.queue_type().supports_graphics())
-                        .map_or(false, |qf| surface.dx11.supports_queue_family(qf))
-                });
+        let mut adapters_dx11 = match instance.dx11 {
```

Comment:
now that all of these blocks look the same, we can consider adding a macro to help manage the repetition (not required for this PR)

Reply:
Yeah, that was my thought too, just didn't want to overcomplicate directly. :)